packages/mobile-top-nav/src/media-menu.js

```diff
@@ -0,0 +1,153 @@
+import { LitElement, html } from 'lit-element';
+
+import './media-slider';
+import './assets/img/media-menu-images';
+import mediaMenuCss from './css/media-menu';
+
+const menuSelection = [
+  {
+    icon: 'web',
+    menu: 'web',
+    label: 'Wayback Machine',
+  },
+  {
+    icon: 'texts',
+    menu: 'texts',
+    label: 'Texts',
+  },
+  {
+    icon: 'video',
+    menu: 'video',
+    label: 'Video',
+  },
+  {
+    icon: 'audio',
+    menu: 'audio',
+    label: 'Audio',
+  },
+  {
+    icon: 'software',
+    menu: 'software',
+    label: 'Software',
+  },
+  {
+    icon: 'images',
+    menu: 'images',
+    label: 'Images',
+  },
+  {
+    icon: 'ellipses',
+    menu: 'more',
+    label: 'More',
+  },
+];
+
+class MediaMenu extends LitElement {
+  static get properties() {
+    return {
+      mediaMenuOpen: { type: Boolean },
+      mediaMenuAnimate: { type: Boolean },
+      mediaSliderOpen: { type: Boolean },
+      mediaSliderAnimate: { type: Boolean },
+      selectedMenuOption: { type: String },
+    };
+  }
+
+  constructor() {
+    super();
+    this.mediaSliderOpen = false;
+    this.mediaSliderAnimate = false;
+    this.selectedMenuOption = '';
+  }
+
+  updated(changedProperties) {
+    const { mediaMenuOpen, mediaSliderOpen } = this;
+    const menuClosed = changedProperties.has('mediaMenuOpen')
+      && changedProperties.get('mediaMenuOpen')
+      && !mediaMenuOpen;
+
+    if (menuClosed && mediaSliderOpen) {
+      this.mediaSliderOpen = false;
+      this.mediaSliderAnimate = false;
+      this.selectedMenuOption = '';
+    }
+  }
+
+  closeMediaSlider() {
+    this.mediaSliderAnimate = true;
+    this.mediaSliderOpen = false;
+    this.selectedMenuOption = '';
+  }
+
+  toggleMediaSlider() {
+    if (!this.mediaSliderOpen) {
+      this.mediaSliderAnimate = true;
+      this.mediaSliderOpen = !this.mediaSliderOpen;
+    }
+  }
+
+  select(mediatype) {
+    const currentSelection = this.selectedMenuOption;
+
+    if (currentSelection === mediatype) {
+      this.closeMediaSlider();
+      return;
+    }
+
+    this.selectedMenuOption = mediatype;
+    this.toggleMediaSlider();
+  }
+
+  get mediaMenuOptionsTemplate() {
+    const buttons = menuSelection.map(({ icon, menu, label }) => {
+      const selected = this.selectedMenuOption === menu ? 'selected' : '';
+      return html`
+        <button class="menu-item ${selected}" @click="${this.select.bind(this, menu)}">
```

Comment:
Using `bind` in a lit-element like this is usually a sign to encapsulate that data in a component. For example, I would make a lit-element for this entire `<button>` block. You can instantiate with a menu option and listen for a click event from it. It would emit a `CustomEvent` when clicked and you handle that in your `select` method. ie:
```
<media-button .menu=menu @selected=${this.select}></media-button>
...
select(e) {
  const menu = e.details.menuOption
}
...
// media-button.js
<button @click=${this.clicked}></button>
clicked() {
    const event = new CustomEvent('selected', {
      detail: { menuOption: this.menuOption },
    });
    this.dispatchEvent(event);
}
```

Reply:
OR you can just do anonymous arrow that sends `menu` to `this.select` when it calls it

Reply:
IMO, that approach puts too much onus on the parent component. Right now he's laying out a subcomponent using a map over `menuSelection` then constructing each subcomponent with a piece from that `menuSelection`. All of that logic can be encapsulated in a child component by passing in the `menuSelection` to it and letting it lay itself out.

Reply:
True, I can get behind the 1 directional data flow. Custom Events seem like the convention for WC to get this done?

Also, this is quite in line with our architecture discussion a few weeks ago around responsibility